### PR TITLE
Show a specific error when upload failed

### LINKED ISSUE #190: Show a specific error when upload failed
Show a specific error when the user failed to submit data on the server. Currently showing only failed form name.

### PR DESCRIPTION
**Pull request by owais-vd**
close #190

### CODE REVIEW

File: src/main/java/com/onaio/steps/helper/UploadFileTask.java [hunk 1]
```diff
@@ -130,4 +134,20 @@ public boolean isAllFieldValid(@Nullable String... fields) {
...
+    @Nullable
+    public String findError(@Nullable ResponseBody errorBody) {
+        String error = null;
+        if (errorBody != null) {
+            try {
+                String rawError = errorBody.string();
+                JSONObject jsonObject = new JSONObject(rawError);
+                error = jsonObject.getString("error");
+            } catch (JSONException | IOException ex) {
+                ex.printStackTrace();
```
Comment (ekigamba) [inline src/main/java/com/onaio/steps/helper/UploadFileTask.java:147]: Log this using Timber

**Review by ekigamba — CHANGES_REQUESTED** [review]

File: src/main/java/com/onaio/steps/utils/CrashlyticsTree.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import timber.log.Timber;
+
+public class CrashlyticsTree extends Timber.DebugTree {
+
+    private static final String CRASHLYTICS_KEY_PRIORITY = "priority";
+    private static final String CRASHLYTICS_KEY_TAG = "tag";
+    private static final String CRASHLYTICS_KEY_MESSAGE = "message";
+
+    @Override
+    protected void log(int priority, @Nullable String tag, @NotNull String message, @Nullable Throwable t) {
```
Comment (ekigamba) [inline src/main/java/com/onaio/steps/utils/CrashlyticsTree.java:26]: change the variable name `t` to `throwable`

**Review by ekigamba — CHANGES_REQUESTED** [review]
Kindly make the changes

File: src/main/java/com/onaio/steps/utils/CrashlyticsTree.java (hunk 2 above)
Comment (owais-vd) [reply src/main/java/com/onaio/steps/utils/CrashlyticsTree.java]: > change the variable name `t` to `throwable`

done

**Review by ekigamba — APPROVED** [review]

**Pull request merged**